sympy/functions/elementary/piecewise.py

```diff
@@ -273,22 +238,23 @@ def eval(cls, *_args):
 
             # collect successive e,c pairs when exprs or cond match
             if newargs:
-                if newargs[-1].expr == expr:
-                    orcond = Or(cond, newargs[-1].cond)
+                e, c = newargs[-1]
+                if e == expr:
+                    orcond = Or(cond, c)
                     if isinstance(orcond, (And, Or)):
                         orcond = distribute_and_over_or(orcond)
-                    newargs[-1] = ExprCondPair(expr, orcond)
+                    newargs[-1] = (expr, orcond)
                     continue
-                elif newargs[-1].cond == cond:
-                    newargs[-1] = ExprCondPair(expr, cond)
+                elif c == cond:
+                    newargs[-1] = (expr, cond)
                     continue
 
-            newargs.append(ExprCondPair(expr, cond))
+            newargs.append((expr, cond))
 
         # some conditions may have been redundant
         missing = len(newargs) != len(_args)
         # some conditions may have changed
-        same = all(a == b for a, b in zip(newargs, _args))
+        same = all(a == b for a, b in zip(newargs, args))
```

Comment:
Why is this checking `args` but for `missing` above it checks `_args`?

Can this not just be `if newargs != args`?

Reply:
input `_args` was changed to `args` and is a tuple while newargs is a list. So instead of converting one type to the other I ust compare them element-wise